airbyte-integrations/bases/base-normalization/dbt-project-template/macros/cross_db_utils/datatypes.sql

```diff
@@ -156,13 +156,117 @@
 {% endmacro %}
 
 {%- macro sqlserver__type_timestamp_with_timezone() -%}
+    datetimeoffset
```

Comment:
We should not be pushing a breaking change to MS SQL normalization. There are multiple cloud and OSS workspaces using it.

Reply:
Return `datetime2` type for MS SQL for both datetime with TZ and without types.

Reply:
two questions:
1. what happens to existing connections, where the table already exists and is using `datetime2`, if we run this updated normalization code and try to set a datetimeoffset value? I.e. does it work correctly, does it throw an error, does it set an incorrect value, etc?
2. _If_ it doesn't work correctly, does that mean we can never support timestamp_with_timezone in mssql? (my understanding is that `datetime2` doesn't include TZ information)
    * or is there some way we can do this backwards-compatibly? Or... if that would be very high-effort, how can we proceed here? I don't think using datetime2 for the rest of eternity is necessarily a good outcome either.

Reply:
There are two cases, if in old version column was `datetime2` datatype and in the new normalization version it becomes `datetimeoffset` datatype:
1. In full refresh case, new tables will be generated with a new `datetimeoffset` type.
2. In incremental sync, all tables will be as it was before (with `datetime2` type).

,

Reply:
@annalvova05 do we have an automated test that verifies behavior for case #2: incremental sync on a connection that was created and synced with older version of normalization will not fail and destination will continue to receive values as `datetime2`?

Reply:
We just have integration tests with incremental sync on a new version, not using two normalization versions.

Reply:
Could you please manually test this scenario? The goal here is to avoid breaking customers and OSS users by publishing new version of normalization image.

Reply:
> There are two cases, if in old version column was `datetime2` datatype and in the new normalization version it becomes `datetimeoffset` datatype:
> 
> 1. In full refresh case, new tables will be generated with a new `datetimeoffset` type.
> 2. In incremental sync, all tables will be as it was before (with `datetime2` type).
> 

These cases were manually tested.